### ISSUE TITLE
Support experimental Netty adaptive allocator 

### ISSUE THREAD

**Issue by franz1981** [quarkusio/quarkus#40934]
### Description

The new Netty release i.e. https://netty.io/news/2024/05/22/4-1-110-Final.html
allow using the new Netty allocator via `-Dio.netty.allocator.type=adaptive` which:
- provide lower memory footprint 
- doesn't uses thread locals (more virtual thread friendly)
- is faster than the default one (thanks to few changes recently implemented by me)

**Comment by cescoffier** [comment]
Thanks for opening this. I wonder if this should become the default or not? 

As a first step we can add an option to switch to this strategy.

**Comment by cescoffier** [comment]
Working on the bump to Netty 4.1.111. So, we would be able to move this forward.

**Comment by franz1981** [comment]
https://github.com/eclipse-vertx/vert.x/issues/5168

This is highly related to this issue and #41880 given that the reason why vertx got a perf hit allocating byte[] with SSL is coming from https://github.com/eclipse-vertx/vert.x/issues/5168#issuecomment-2258173211

**Comment by cescoffier** [comment]
Waiting for upstream release.

**Comment by cescoffier** [comment]
@franz1981 do you know the status on this? 
I'm truing to follow the progress on the vert.x side (as we are integrating 4.5.12 at the moment), but I just stumbled on https://github.com/eclipse-vertx/vert.x/issues/5460.

\CC  @jponge

**Comment by franz1981** [comment]
Yep, I was waiting the Key cloak folks to validate the performance for them - including the ability of pool heap buffers with SSL

**Comment by franz1981** [comment]
In relation of the adaptive pool we are in a good shape for Netty 4.2 already - just few things to solve perf wise, but not evident bugs so far

**Comment by geoand** [comment]
Is this something we are planning on allowing for Quarkus 4?

**Comment by cescoffier** [comment]
I think we went all in on this. @jponge / @franz1981 ?

**Comment by jponge** [comment]
Let's hear back from @franz1981

**Comment by franz1981** [comment]
I'm still working on the new version of this :) regardless 4.2 will ship it by default so this issue is not needed per se

**Comment by jponge** [comment]
Thanks @franz1981
